src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -390,6 +397,50 @@ pg_get_tableschemadef_string(Oid tableRelationId, IncludeSequenceDefaults
 			}
 #endif
 
+			if (attributeForm->attidentity && includeIdentityDefaults)
+			{
+				bool missing_ok = false;
+				Oid seqOid = getIdentitySequence(RelationGetRelid(relation),
+												 attributeForm->attnum, missing_ok);
+
+				char *sequenceName = generate_qualified_relation_name(seqOid);
+
+				if (includeIdentityDefaults == INCLUDE_IDENTITY_AS_SEQUENCE_DEFAULTS)
+				{
+					if (pg_get_sequencedef(seqOid)->seqtypid != INT8OID)
+					{
+						appendStringInfo(&buffer,
+										 " DEFAULT worker_nextval(%s::regclass)",
+										 quote_literal_cstr(sequenceName));
+					}
+					else
+					{
+						appendStringInfo(&buffer, " DEFAULT nextval(%s::regclass)",
```

Comment:
I'm kind of wondering whether we could always do worker_nextval, and then worker_nextval can decide what to do in case of int8 vs. int4 vs. int2 (e.g. it can call regular nextval in case of int8, and error out in other cases). 

That way, we have a bit more freedom to adjust sequence number generation policy later (e.g. we could introduce a setting that changes the policy to getting numbers 1-by-1 from the coordinator, or in batches).

I'm trying to remember why we did not use worker_nextval for bigint sequences, maybe as an expedient, or because we already had DEFAUL nextval columns prior to introducing worker_nextval.

Reply:
ah, we probably did not do this because we have specific tweaks for nextval() in the planner, e.g. in case of insert into .. select.